### DISCUSSION TITLE
Ability for Bluetooth "bridge"? for controllers?

### DISCUSSION THREAD

**Discussion by Dark1886** [tttapa/Control-Surface#916] [General]
I'm currently imagining a scenario where I would have a footswitch board for controlling MIDI presets/effects enabling and a second board with sliders/knobs for other items such as gain. With this current build is there any way to use bluetooth to connect the two and allow only one of them to be connected to the host PC?

I admit the main way to do this is probably easiest to just connect the two by some cable to handle all of this, but I think some sort of BLE bridge would be cool.

**Comment by tttapa** [comment]
Unfortunately, this is currently not supported, Control Surface only supports server mode, not client mode, and I'm afraid I won't have much time in the near future to implement it either :( But this is certainly something you could pull off, there are GATT _server_ and GATT _client_ examples in the ESP-IDF repository: https://github.com/espressif/esp-idf/tree/master/examples/bluetooth/bluedroid/ble.

Keep in mind that Control Surface's BLE implementation is based on the examples from around five years back (ESP-IDF v3.2), there might be parts that need an update. You can find links to the original examples I used in https://github.com/tttapa/Control-Surface/tree/main/src/MIDI_Interfaces/BLEMIDI/ESP32#esp32-midi-over-bluetooth-low-energy.

That being said, if you just need to send MIDI data from one ESP32 to another, there are certainly simpler protocols than fully compliant MIDI over BLE (e.g. a WiFi TCP socket or a Bluetooth serial port).
